x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/session/EsqlSession.java

```diff
@@ -292,21 +293,25 @@ private <T> void preAnalyze(
         var unresolvedPolicies = preAnalysis.enriches.stream()
             .map(e -> new EnrichPolicyResolver.UnresolvedPolicy((String) e.policyName().fold(), e.mode()))
             .collect(Collectors.toSet());
-        final Set<String> targetClusters = enrichPolicyResolver.groupIndicesPerCluster(
+        final Set<String> clusters = enrichPolicyResolver.groupIndicesPerCluster(
             preAnalysis.indices.stream()
                 .flatMap(t -> Arrays.stream(Strings.commaDelimitedListToStringArray(t.id().index())))
                 .toArray(String[]::new)
         ).keySet();
+
+        // key: cluster alias; value = skip_unavailable setting (false for local cluster)
+        Map<String, Boolean> targetClusters = new HashMap<>();
+        for (String alias : clusters) {
+            targetClusters.put(alias, executionInfo.isSkipUnavailable(alias));
+        }
```

Comment:
Can this be just 
```
        Map<String, Boolean> targetClusters = clusters.stream().collect(Collectors.toMap(alias -> alias, executionInfo::isSkipUnavailable));
```

Reply:
Based on research that @original-brownbear has done, it is often better to not use Java stream, especially for short loops - there is a lot of overhead. I did however, change line 303 to use Maps.newHashMapWithExpectedSize since we know the exact size of the new map. Changed in next push.